lib/sidekick/sidekick.ex

```diff
@@ -1,75 +1,88 @@
 defmodule Sidekick do
-  @spec start(atom, [{atom, any}]) :: {:error, any} | {:ok, atom, pid}
+  @spec start(atom, [Parent.child_spec()]) :: :ok | {:error, :already_started | :boot_error}
   def start(node_name \\ :docker, children) do
-    parent_node = Node.self()
-    node = node_host_name(node_name)
+    ensure_distributed!()
 
-    case Node.ping(node) do
-      :pang -> wait_for_sidekick(node, parent_node, children)
-      :pong -> {:error, "Sidekick node #{node} is already alive"}
-    end
+    node = :"#{node_name}@#{hostname()}"
+
+    if Node.ping(node) == :pong,
+      do: {:error, :already_started},
+      else: start_node(node, children)
   end
 
-  defp call(node, module, function, args) do
-    :rpc.block_call(node, module, function, args)
+  defp ensure_distributed! do
+    System.cmd("epmd", ["-daemon"])
+
+    node_name = :crypto.strong_rand_bytes(16) |> Base.encode32(padding: false, case: :lower)
+
+    case :net_kernel.start([:"#{node_name}@127.0.0.1"]) do
```

Comment:
Should we hardcode hostname here?

Reply:
This is a convenience auto-switch to distribution, which is needed only in dev/test. I think 127.0.0.1 is a reasonable hardcoded choice that should work on all platforms, but I'm open to counter arguments :-)